File: app/src/main/java/com/doubleclick/spactrumscanner/api/RetrofitInstance.java
package com.doubleclick.spactrumscanner.api;

import static com.doubleclick.spactrumscanner.utils.Constant.BASE_URL;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created By Eslam Ghazy on 1/3/2023
 */
public class RetrofitInstance {

    private static RetrofitInstance INSTANCE;

    public RetrofitInstance() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
//        postInterface = retrofit.create(API.class);
    }

    public static RetrofitInstance getINSTANCE() {
        if (null == INSTANCE) {
            INSTANCE = new RetrofitInstance();
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/doubleclick/spactrumscanner/RetrofitInstance.kt
package com.doubleclick.spactrumscanner

import com.doubleclick.spactrumscanner.api.API
import com.doubleclick.spactrumscanner.utils.Constant.BASE_URL
import com.google.gson.GsonBuilder
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created By Eslam Ghazy on 1/3/2023
 */
object RetrofitInstance {

    private val retrofit by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create(GsonBuilder().setLenient().create()))
            .build()
    }

    val api: API by lazy {
        retrofit.create(API::class.java)
    }

}